### ISSUE TITLE
inconsistent behavior of slicing in certain model file

### ISSUE THREAD

**Issue by youkaichao** [huggingface/safetensors#495]
### System Info

safetensors                              0.4.2

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Reproduction

```python
with safe_open("/home/gcpuser/.cache/huggingface/hub/models--meta-llama--Llama-2-7b-hf/snapshots/01c7f73d771dfac7d292323805ebc428287df4f9/model-00001-of-00002.safetensors", framework="pt") as f:
    tensor_slice = f.get_slice("model.layers.0.input_layernorm.weight")
    tensor = f.get_tensor("model.layers.0.input_layernorm.weight")
    sliced = tensor_slice[tuple()]
    ground_truth = tensor[tuple()]
assert ground_truth.shape == sliced.shape # error!
```

the file is downloaded from https://huggingface.co/NousResearch/Llama-2-7b-hf/tree/main .

### Expected behavior

`sliced` should be a full tensor rather than an empty tensor.

**Comment by Narsil** [comment]
What is `tensor[tuple()]` trying to achieve ?

Can you provide a link into torch documentation showcasing the intent there ?

`tensor[list()] == torch.empty()` , so I'd argue that the tuple indexing is unintended behavior within torch itself.

Regardless I'd need a solid foundation for the behavior for it to change.


`tensor[list()]` being empty can be understood as `tensor[ [0, 0, 1] ]` is asking for repeated first row then second row. In that sense empty list means asking for no row, therefore empty.
I am no seeing any reason why tuple should be treated differently.

**Comment by youkaichao** [comment]
I think this behavior is inherited from numpy:

```python
import numpy as np
a = np.arange(10)
a[tuple()] # array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
a[list()] # array([], dtype=int64)
```

Why tuple and list are treated differently?

My understanding is, multi-dimentional slicing is essentially a `tuple`: `tensor[a, b, c]` in python, is `tensor.__getitem__(slice)`, where `slice = (a, b, c)`. Since `tensor[a, b]` means `tensor[a, b, :]`, it is natural to think `tensor[tuple()]` is `tensor[:, :, :]`, that is, a view of the whole tensor.

For reference, numpy has a detailed note for empty tuple slicing: https://numpy.org/doc/stable/user/basics.indexing.html#detailed-notes

Meanwhile, a list itself, is a single dimentional slicing: `tensor[[1, 2, 3]]` in python, is `tensor.__getitem__(slice)`, where `slice = [1, 2, 3]`. Since `tensor[[1, 2, 3]]` is stacking `tensor[1], tensor[2], tensor[3]`, it is natural to think `tensor[list()]` is an empty tensor.

**Comment by Narsil** [comment]
@youkaichao You make a great point here actually, I stand corrected.

This behavior was fixed in 0.4.3 actually (I do not remember what has changed tbh).
Can you confirm it works with 0.4.3 ? I will also add a specific test in CI so that this behavior remains checked for.

**Comment by youkaichao** [comment]
That's good to know. Thanks for the fix!
